[WPE] Replace strcmp in WPEDisplayWayland

spenap opened the pull request:
#### 281df01f819767638ea2dfa81c3165e64dab691e
<pre>
[WPE] Replace strcmp in WPEDisplayWayland
<a href="https://bugs.webkit.org/show_bug.cgi?id=289891">https://bugs.webkit.org/show_bug.cgi?id=289891</a>

Reviewed by Adrian Perez de Castro.

Use ASCIILiteral comparisons instead of strcmp in WPEDisplayWayland.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:

Canonical link: <a href="https://commits.webkit.org/292263@main">https://commits.webkit.org/292263@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/48cbee9dd9c6449b80f68098b1e4c48d44fab138

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95435 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15038 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4906 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100485 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45941 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15325 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23471 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72794 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30058 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98438 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11468 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86155 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53125 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11180 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45277 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81362 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3980 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102522 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22485 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16421 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81810 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-on-non-root.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22734 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82164 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81182 "Found 1 new API test failure: /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/replace-get-all-cookies (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20325 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25757 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3183 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15804 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22454 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27593 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22113 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25588 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23853 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
```diff
@@ -219,32 +220,34 @@ const struct wl_registry_listener registryListener = {
-        if (!std::strcmp(interface, "wl_compositor"))
+        auto interfaceSpan = unsafeSpan(interface);
+
+        if (!equalSpans(interfaceSpan, "wl_compositor"_span))
```
I guess we need to invert the logic, I assume `equalSpans`, that returns a bool, returns true if spans are equal and false otherwise.

spenap replied:
oooops. And I should had seen it, I was following another PR where they had done that. Thanks, Carlos, sorry

aperezdc commented on Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
```diff
@@ -219,32 +220,34 @@ const struct wl_registry_listener registryListener = {
-        if (!std::strcmp(interface, "wl_compositor"))
+        auto interfaceSpan = unsafeSpan(interface);
+
+        if (!equalSpans(interfaceSpan, "wl_compositor"_span))
```
I would use `StringView::fromLatin1` and comparing against `ASCIILiteral` literals:

```suggestion
        const auto interfaceName = StringView::fromLatin1(interface);

        if (interfaceName == "wl_compositor"_s)
```

I find using the overloaded `==` operator in this case clearer that adding the `equalSpans()` calls around.

spenap committed:
[WPE] Replace strcmp in WPEDisplayWayland
https://bugs.webkit.org/show_bug.cgi?id=289891

Reviewed by Adrian Perez de Castro.

Use ASCIILiteral comparisons instead of strcmp in WPEDisplayWayland.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:

Canonical link: https://commits.webkit.org/292263@main
